v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -448,6 +468,31 @@ func (r *azureDeploymentReconcilerInstance) handleCreatePollerSuccess(ctx contex
 	return ctrl.Result{}, nil
 }
 
+func (r *azureDeploymentReconcilerInstance) postReconciliationCheck(ctx context.Context) (extensions.PostReconcileCheckResult, error) {
+	// Create a checker for access to the extension point, if required
+	checker, extensionFound := extensions.CreatePostReconciliationChecker(r.Extension)
+	if !extensionFound {
+		// No extension found, nothing to do
+		return extensions.PostReconcileCheckResultSuccess(), nil
+	}
+
+	// We also need to have our owner, it too with an up-to-date status
+	owner, ownerErr := r.ResourceResolver.ResolveOwner(ctx, r.Obj)
+	if ownerErr != nil {
+		// We can't obtain the owner, so we can't run the extension
+		return extensions.PostReconcileCheckResult{}, ownerErr
+	}
```

Comment:
What about resources that don't have an owner that's an ASO resource?

Reply:
Is there any resource that does not have an owner except `ResourcrGroups` ?

Reply:
`ResolveOwner()` returns a `genruntime.ARMMetaObject`, so it can only return a resource that exists in the cluster - and there's no requirement that people create their `ResourceGroup` in the cluster.

For example, in ARM a `BatchAccount` is created within a `ResourceGroup`, but in ASO you could just create the `BatchAccount`, putting it inside a ResourceGroup that already exists.

Reply:
Do we allow this scenario where the owner does not exist in ASO? I think it fails while we try to `Claim` the resource